src/go/rpk/pkg/config/weak_test.go

```diff
@@ -0,0 +1,429 @@
+			data:   `wi: "-23414"`,
+			expInt: -23414,
+			expErr: false,
+		},
+		{
+			name:   "should unmarshal string:231231",
+			data:   `wi: "231231"`,
+			expInt: 231231,
+			expErr: false,
+		},
+		{
+			name:   "should unmarshal bool:true",
+			data:   `wi: true`,
+			expInt: 1,
+			expErr: false,
+		},
+		{
+			name:   "should unmarshal bool:false",
+			data:   `wi: false`,
+			expInt: 0,
+			expErr: false,
+		},
+		{
+			name:   "should error with not-numeric strings",
+			data:   `wi: "123foo234"`,
+			expErr: true,
+		},
+		{
+			name:   "should error with float numbers",
+			data:   `wi: 123.234`,
+			expErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ts := testWeakInt{}
+			err := yaml.Unmarshal([]byte(test.data), &ts)
+
+			gotErr := err != nil
+			if gotErr != test.expErr {
+				t.Errorf("input %q: got err? %v, exp err? %v; error: %v",
+					test.data, gotErr, test.expErr, err)
+				return
+			}
+			if test.expErr {
+				return
+			}
+
+			if int(ts.Wi) != test.expInt {
+				t.Errorf("input %q: got %v, expected %v",
+					test.data, ts.Wi, test.expInt)
+				return
+			}
+		})
+	}
+}
+
+func TestWeakString(t *testing.T) {
+	type testWeakString struct {
+		Ws weakString `yaml:"ws"`
+	}
+	tests := []struct {
+		name   string
+		data   string
+		expStr string
+		expErr bool
+	}{
+		{
+			name:   "should unmarshal normal string types",
+			data:   `ws: "hello world"`,
+			expStr: "hello world",
+			expErr: false,
+		},
+		{
+			name:   "should unmarshal bool:true",
+			data:   "ws: true",
+			expStr: "1",
+			expErr: false,
+		},
+		{
+			name:   "should unmarshal bool:false",
+			data:   "ws: false",
+			expStr: "0",
+			expErr: false,
+		},
+		{
+			name:   "should unmarshal base10 number:231231",
+			data:   "ws: 231231",
+			expStr: "231231",
+			expErr: false,
+		},
+		{
+			name:   "should unmarshal float number:231.231",
+			data:   "ws: 231.231",
+			expStr: "231.231",
+			expErr: false,
+		},
+		{
+			name:   "should error with unsupported types",
+			data:   "ws: \n  - 231.231",
+			expErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ts := testWeakString{}
+			err := yaml.Unmarshal([]byte(test.data), &ts)
+
+			gotErr := err != nil
+			if gotErr != test.expErr {
+				t.Errorf("input %q: got err? %v, exp err? %v; error: %v",
+					test.data, gotErr, test.expErr, err)
+				return
+			}
+			if test.expErr {
+				return
+			}
+
+			if string(ts.Ws) != test.expStr {
+				t.Errorf("input %q: got %v, expected %v",
+					test.data, ts.Ws, test.expStr)
+				return
+			}
+		})
+	}
+}
+
+func TestNamedSocketAddressArray(t *testing.T) {
+	type testNamedSocketAddressArray struct {
+		Sockets NamedSocketAddresses `yaml:"test_api"`
+	}
+	tests := []struct {
+		name   string
+		data   string
+		expArr []NamedSocketAddress
+		expErr bool
+	}{
+		{
+			name: "it should unmarshal a single namedSocketAddress as a slice",
```

Comment:
Perhaps drop `it should unmarshal a`, per comments above? `single named socket address`

Also while I'm fwiw'ing, not all tests need names, though it does make it easier to tell which exact test failed. Before `t.Run` existed, the pattern was to just print the loop number that failed.